File: samples/hello_world/java/HelloWorld.java
public class HelloWorld {
  private static int _id_seq = 0;

  private int _id;

  public static int hello = 1;

  public HelloWorld() {
    this._id = (_id_seq++);
  }

  public void say() {
    System.out.println("Hello, world! My is id " + _id);
  }

  public void destroy() {
    System.out.println("Goodbye, world! My id is " + _id);
  }

  private void private_method() {
      hello = 2;
      System.out.println(hello);
  }
}

File: samples/hello_world/cpp/hello_world.cc
#include <iostream>
#include "hello_world.h"

int HelloWorld::_id_seq = 0;

int HelloWorld::public_variable = 0;

void HelloWorld::private_method() {
  std::cout << "prrr" << std::endl;
}

HelloWorld::HelloWorld() {
  this->_id = (HelloWorld::_id_seq++);
}

void HelloWorld::destroy() {
  std::cout << "Goobdye, world! My id is " << this->_id << std::endl;
}

void HelloWorld::say() {
  std::cout << "Hello, world! My id is " << this->_id << std::endl;
}

